app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt

```diff
@@ -0,0 +1,166 @@
+    }
+
+    fun fromLocalDateTimeToTimeStamp(date: LocalDateTime): Long {
+        return date.toEpochSecond(
+            ZoneId.systemDefault()
+                .rules.getOffset(Instant.now())
+        )
+    }
+
+    fun formatDate(date: Long): String {
+        val localDateTime = LocalDateTime.ofInstant(
+            Instant.ofEpochSecond(date),
+            ZoneId.systemDefault()
+        )
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(
+            "dd.MM.yyy",
+            Locale.getDefault()
+        )
+        return formatter.format(localDateTime)
+    }
+
+    fun formatTime(date: Long): String {
+        val localDateTime = LocalDateTime.ofInstant(
+            Instant.ofEpochSecond(date),
+            ZoneId.systemDefault()
+        )
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(
+            "HH:mm",
+            Locale.getDefault()
+        )
+        return formatter.format(localDateTime)
+    }
+
+    suspend fun <T, R> makeRequest(
+        request: suspend () -> Response<T>,
+        onSuccess: suspend (body: T) -> R
+    ): R {
+        try {
+            val response = request()
+            if (!response.isSuccessful) {
+                throw ApiException(response.code(), response.message())
+            }
+            val body =
+                response.body() ?: throw ApiException(response.code(), response.message())
+            return onSuccess(body)
+        } catch (e: IOException) {
+            throw ServerException
+        } catch (e: Exception) {
+            e.printStackTrace()
+            throw UnknownException
+        }
+    }
+
+    // TODO переделать в ресурсы, форматированную строку сделай
+    fun generateShortUserName(lastName: String, firstName: String, middleName: String): String {
+        return "$lastName ${firstName.first().uppercase()}. ${middleName.first().uppercase()}."
+    }
+
+    fun fullUserNameGenerator(lastName: String, firstName: String, middleName: String): String {
+        return "$lastName $firstName $middleName"
+    }
+
+    fun isOnline(context: Context): Boolean {
+        var result = false // Returns connection. false: none; true: mobile data or wifi
+        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            cm?.run {
+                cm.getNetworkCapabilities(cm.activeNetwork)?.run {
+                    when {
+                        hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> {
+                            result = true
+                        }
+                        hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> {
+                            result = true
+                        }
+                        hasTransport(NetworkCapabilities.TRANSPORT_VPN) -> {
+                            result = true
+                        }
+                    }

```

Comment:
А так случайно не работает?
```suggestion
                    result = hasTransport(NetworkCapabilities.TRANSPORT_WIFI or NetworkCapabilities.TRANSPORT_CELLULAR or NetworkCapabilities.TRANSPORT_VPN)
```